analytics_nudge.md

```diff
@@ -0,0 +1,57 @@
+    
+### How do we nudge, and how does the user choose?
+
+We should nudge the user in the Web UI, and allow them to select "opt-in" or "opt-out" via button click.
+
+When Tilt determines that we need to nudge the user, we'll send that information along to the Web UI (as part of the Webview json) and render an unobtrusive nudge in sidebar (area B in this image), though in the status bar (C) or in the nav bar (A, though this conflicts with Sail share info) are also possibilities. (Particularly if we nudge when the user gets their first resource up and running, seems safe to assume that we'll have space to spare in the sidebar.)
+
+![possible places in web UI to put analytics nudge](images/web-ui-where-nudge.png) 
+
+The nudge will consist of some copy tbd, and buttons for "opt in" and "opt out". Clicking one of those buttons sends an HTTP request back to the HUD server, which propagates it to the relevant analytics code (which writes the user's choice to disk, as normal). (If something goes wrong with this process, we log the error and replace the buttons with a "something went wrong, try again later" message.)
+
+Copy TDB: either something straight-forward like "opt into analytics to help us better support you!" or a reference to the condition under which we nudged (see above) like, "Congrats on your first Tilt deploy 🎉 want to send analytics to help us better support you?" Will contain a "read more" link to a page on our website about privacy policy, how we use metrics, what metrics we send, etc.
+
+#### Alternatives considered
+1. nudge in the TUI.
+    * we don't want to be adding more functionality to the TUI at this time, as we're trying to move towards the Web UI as the way of the future
+2. have the user run the existing `tilt analytics` command
+    * having to switch back to their terminal interrupts the user's flow with Tilt and makes our analytics seem extra annoying -- not to mention that it's especially weird to take a user _out_ of the Web UI when we're trying to establish its primacy 
+3. nudge in the "Alerts" pane
+    * it's appealing to use the Alerts pane to mark "opt in/out of analytics" as an outstanding action item, but it seems too agressive.
+4. a "snooze"/"ask me again later" feature
+    * not only would we have to implement something to track when we last nudged/when to nudge again, I expect that that the bigger of a deal we make opting in/out of analytics, the bigger a deal it will seem to the _user_, and that the more chance we give them to put it off, the more they _will_ put it off.
+
+## Risks
+* if we mess something up technically or implement this nudge in a way that doesn't resonate with users (e.g. they don't see it, they get annoyed and hit "no", we don't provide them enough information and they hit "no" out of fear), we risk a gap in metrics. This makes it harder for us to know what's going on with our product and could negatively affect fundraising
```

Comment:
you identify this as a major risk, but i didn't see any discussion on how we address this? maybe we send an event every time someone opts in or out of analytics?

Reply:
...yeah that's fair. not sure what you're suggesting about sending events, though?